### PR TITLE
185259/Validate-all-files

### PR DESCRIPTION
**Pull request by Kirsten1822**
Ticket link: https://dev.azure.com/defragovuk/DEFRA-FFC/_sprints/taskboard/DEFRA-FFC%20Payments%20Team/DEFRA-FFC/Payment%20Service/Sprint%2006?workitem=185259

### CODE REVIEW

File: test/integration/verify.test.js [hunk 1]
```diff
@@ -179,28 +199,75 @@ describe('verify batch content', () => {
...
+    await expect(verifyBatch).rejects.toThrow()
+  })
+
+  test('verify batch throws error if checksum control file missing', async () => {
+    await uploadBlob(ORIGINAL_BATCH_BLOB_NAME, VALID_CONTENT)
+    await uploadBlob(ORIGINAL_CTL_BATCH_BLOB_NAME, EMPTY_CONTENT)
+    await uploadBlob(ORIGINAL_CHECKSUM_BLOB_NAME, VALID_HASH)
+
+    await expect(verifyBatch).rejects.toThrow()
+  })
```
Comment (johnwatson484) [inline test/integration/verify.test.js]: These three tests don't actually do anything.  They upload files, then check that calling verifyBatch results in an error, which it always will.  It doesn't actually call the `pollInbound` function.

But thinking about this, I think we've been going the wrong way about validating this behaviour.  This is an integration test, so we should be looking at outcomes, but we've been focusing on how can we check that `verifyBatch` throws an error.

If a file is missing, then what we should see if the original files still in the inbound folder.  So I think a simpler way of writing these tests would be, get rid of what I suggested earlier, ie the `verifyBatch` stuff above.

Then call poll inbound here and then get the files in the inbound folder the same way the tests below do and check that each of the three files uploaded are still there unchanged.

That seems a better test.

**Review by johnwatson484 — COMMENTED** [review]
Only one comment regarding some of the tests.  

I think there's a better way we can test them to get the confidence we need.

File: test/integration/verify.test.js (hunk 1 above)
Comment (Kirsten1822) [reply test/integration/verify.test.js]: I can take a look at doing that yeah - thanks

File: test/integration/verify.test.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
 const { BlobServiceClient } = require('@azure/storage-blob')
+
 jest.mock('../../app/config/verify', () => ({ totalRetries: 1 }))
 const createHash = require('../../app/verify/create-hash')
+
+jest.mock('../../app/verify', () => {
+  const originalModule = jest.requireActual('../../app/verify')
+  return originalModule
+})
+const verifyBatch = require('../../app/verify')
```
Comment (johnwatson484) [inline test/integration/verify.test.js]: This can all be removed now

**Review by johnwatson484 — APPROVED** [review]

**Pull request merged**